usb/core.py

```diff
@@ -1289,3 +1289,18 @@ def show_devices(verbose=False, **kwargs):
             strings += "%s\n\n" % str(device)
 
     return _DescriptorInfo(strings)
+
+def device_from_fd(fd):
+    import usb.backend.libusb1 as libusb1
+
+    backend = libusb1.get_backend()
+    if not backend:
+        raise NoBackendError('No backend available')
+
+    dev = backend.get_device_from_fd(fd)
+
+    # create device
+    device = Device(dev, backend)
+    device._ctx.handle = dev
```

Comment:
I find it ugly, too, because it uses the private members. But it is necessary to work correctly (not sure why because I tested it long ago).
I simply tried to replicate the same setup steps but also wanted to correctly separate what has to happen where, the interfaces, the wrapper code etc.
As for the `_DeviceHandle`, I'm not sure. Again, too long ago but I might have missed it.
And I wanted it self-contained because as I wrote this, this was mostly intended for Termux. Other OS might not have to go the way of using the `fd` and can just use the path?